lib/samples/apply_simple_renderer_to_feature_layer/apply_simple_renderer_to_feature_layer_sample.dart

```diff
@@ -42,11 +43,11 @@ class _ApplySimpleRendererToFeatureLayerSampleState
     _map.operationalLayers.add(_featureLayer);
     _map.initialViewpoint = Viewpoint.fromCenter(
       ArcGISPoint(
-        x: -9177250,
-        y: 4247200,
+        x: -9177343,
+        y: 4247283,
         spatialReference: SpatialReference.webMercator,
       ),
-      scale: 9500,
+      scale: 4750,
     );
     _mapViewController.arcGISMap = _map;
```

Comment:
We've established a pattern going forward that we'll do this setup within `_onMapViewReady`. It makes the story cleaner for those using the samples, which is sometimes a "quirk" of our samples - to balance absolute best practice with being very readable. Even though having it in `initState` is completely valid here. (see this discussion for background https://github.com/Esri/arcgis-maps-sdk-flutter-samples/pull/35#discussion_r1654467239)

Reply:
We'll also need to add in the `setState(() => _ready = true)` mechanism, to protect the button callbacks from hitting the `late FeatureLayer _featureLayer` before it gets set.

Actually I really like using the `Visibility(visible: _ready, child: overlayControls(context))` pattern because it protects all the callbacks in one fell swoop. Maybe that can be a pattern for all samples that have `late` variables used in callbacks?

Reply:
Added a Visibility widget with overlay controls that block user input.

Reply:
I quite like the use of the progress indicator, as this can be useful for samples that are downloading data too, and makes it clear something is happening. It is something noted in an issue about the offline data. I also used a progress indicator in show service area but looking at it again I think that sample could be improved with what you're saying @PaulAllanSturm, as I do `_ready` checks in multiple places, which is probably not necessary - I could probably capture that in a higher level widget.

So maybe we can make the pattern include a `CircularProgressIndicator`.

Reply:
Oh - and for `Visibility(visible: _ready, child: overlayControls(context))` - @PaulAllanSturm are you meaning actually using `overlayControls` as the general rule? I personally prefer using the widget tree directly unless it's getting really long. But definitely agree on the Visibility pattern.

Reply:
For `CircularProgressIndicator`, that makes sense for when there's a download going on, plus for any long operation. As for the gap from creating the widget until the map is ready, it might be overkill, especially if it is visually unpleasant. There might be enough visual feedback already, as you go from a clear map to seeing the buttons appear.

`visible: _ready` is the important part of the `Visibility` pattern. `overlayControls()` would just be when we want to factor out a busy `build()` method.

Reply:
Sounds good to me 👍

Reply:
This page load happens so fast, we don't usually see the overlay controls. This does contain the `CircularProgressIndicator`. In the case that the overlay controls are around for a bit, it is good to have something moving indicating that the app didn't just lock up.

Reply:
Oh, in my head I was thinking of "overlay controls" to mean the buttons that operate the various functionality (they're overlaying the MapView). You were thinking of "overlay controls" as the CircularProgressIndicator that overlays the buttons.

Yeah if the onMapViewReady has any `awaits` for network resources, there should be a CircularProgressIndicator in there somehow. Maybe there's a more appropriate Widget than Visibility for that scenario. I think there is, but I'm not going to remember it until after I submit this comment.

Reply:
Maybe IndexedStack https://api.flutter.dev/flutter/widgets/IndexedStack-class.html

Reply:
I'd be happy to move this discussion to a separate issue so that it doesn't have to hold up the PR, if you agree.

Reply:
I've created discussion #1908 in the Flutter repo to continue this conversation.